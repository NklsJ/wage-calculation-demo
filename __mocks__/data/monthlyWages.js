
export default [
  {
    month: 1,
    year: 2015,
    workers: [
      {
        id: 1,
        fullname: 'Foo Bar',
        days: [
          {
            date: '1.1.2015',
            eveningHours: 0,
            hours: 8,
            overtimeHours: 0,
            shifts: [
              {
                start: '8:00',
                end: '16:00',
              },
            ],
            wage: 34,
          },
          {
            date: '2.1.2015',
            eveningHours: 0,
            hours: 10,
            overtimeHours: 2,
            shifts: [
              {
                start: '6:00',
                end: '10:00',
              },
              {
                start: '12:00',
                end: '18:00',
              },
            ],
            wage: 44.63,
          },
        ],
      },
      {
        id: 2,
        fullname: 'Bar Baz',
        days: [
          {
            date: '1.1.2015',
            eveningHours: 0,
            hours: 8,
            overtimeHours: 0,
            shifts: [
              {
                start: '8:00',
                end: '16:00',
              },
            ],
            wage: 34,
          },
          {
            date: '2.1.2015',
            eveningHours: 6,
            hours: 10,
            overtimeHours: 2,
            shifts: [
              {
                start: '00:00',
                end: '10:00',
              },
            ],
            wage: 52.13,
          },
        ],
      },
      {
        id: 3,
        fullname: 'Asd Das',
        days: [
          {
            date: '1.1.2015',
            eveningHours: 2,
            hours: 12,
            overtimeHours: 4,
            shifts: [
              {
                start: '4:00',
                end: '16:00',
              },
            ],
            wage: 58.81,
          },
          {
            date: '3.1.2015',
            eveningHours: 8,
            hours: 12,
            overtimeHours: 4,
            shifts: [
              {
                start: '20:00',
                end: '21:00',
              },
              {
                start: '23:00',
                end: '10:00',
              },
            ],
            wage: 66.31,
          },
        ],
      },
    ],
  },
]
