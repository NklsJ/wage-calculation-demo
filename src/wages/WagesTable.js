// @flow
import React from 'react'

const WagesTable = () => {
  return (
    <div>
      <h3>Monthly Wages 03/2019</h3>
      <table>
        <tbody>
          <tr>
            <td>1.</td>
            <td>Foo Bar</td>
            <td>$1337.00</td>
          </tr>
          <tr>
            <td>2.</td>
            <td>Baz Bar</td>
            <td>$112.00</td>
          </tr>
          <tr>
            <td>3.</td>
            <td>Asd Das</td>
            <td>$313.00</td>
          </tr>
        </tbody>
      </table>
    </div>
  )
}

export default WagesTable
