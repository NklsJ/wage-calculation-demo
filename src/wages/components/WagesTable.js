// @flow
import React from 'react'
import WageRow from './WageRow'
import css from './WagesTable.module.css'
import {type Worker} from '../types'

type Props = {
  workers: Array<Worker>,
}

const WagesTable = ({workers}: Props) => {
  return (
    <div className={css.wageTable}>
      {workers.sort((a, b) => a.id - b.id)
        .map((item, index) => {

          if (
            !item.id ||
              !item.fullname ||
              !item.days
          ) {
            return null
          }

          return <WageRow key={index} worker={item}/>
        })}
    </div>
  )
}

export default WagesTable
