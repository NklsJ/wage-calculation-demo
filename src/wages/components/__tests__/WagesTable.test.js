import React from 'react'
import {shallow} from 'enzyme'

import WagesTable from '../WagesTable'
import monthlyWages from '../../../../__mocks__/data/monthlyWages'

describe('WagesTable component: ', () => {

  it('Should render correctly', () => {
    const component = shallow(<WagesTable workers={monthlyWages[0].workers} />)
    expect(component).toMatchSnapshot()
  })

})
