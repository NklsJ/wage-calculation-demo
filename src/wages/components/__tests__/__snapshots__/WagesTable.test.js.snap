// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WagesTable component:  Should render correctly 1`] = `
<div
  className="wageTable"
>
  <WageRow
    key="0"
    worker={
      Object {
        "days": Array [
          Object {
            "date": "1.1.2015",
            "eveningHours": 0,
            "hours": 8,
            "overtimeHours": 0,
            "wage": 34,
          },
          Object {
            "date": "2.1.2015",
            "eveningHours": 0,
            "hours": 10,
            "overtimeHours": 2,
            "wage": 44.63,
          },
        ],
        "fullname": "Foo Bar",
        "id": 1,
      }
    }
  />
  <WageRow
    key="1"
    worker={
      Object {
        "days": Array [
          Object {
            "date": "1.1.2015",
            "eveningHours": 0,
            "hours": 8,
            "overtimeHours": 0,
            "wage": 34,
          },
          Object {
            "date": "2.1.2015",
            "eveningHours": 6,
            "hours": 10,
            "overtimeHours": 2,
            "wage": 52.13,
          },
        ],
        "fullname": "Bar Baz",
        "id": 2,
      }
    }
  />
  <WageRow
    key="2"
    worker={
      Object {
        "days": Array [
          Object {
            "date": "1.1.2015",
            "eveningHours": 2,
            "hours": 12,
            "overtimeHours": 4,
            "wage": 58.81,
          },
          Object {
            "date": "3.1.2015",
            "eveningHours": 8,
            "hours": 12,
            "overtimeHours": 4,
            "wage": 66.31,
          },
        ],
        "fullname": "Asd Das",
        "id": 3,
      }
    }
  />
</div>
`;
